### ISSUE TITLE
dict entries with a None value are not added to dict

### ISSUE THREAD

**Issue by plajjan** [actonlang/acton#1495]
### Acton Version

0.16.0.20230912.13.31.35

### Steps to Reproduce

dict entries with a None value are not added to dict

```python
actor main(env):
    a = {}
    a["foo"] = 1
    a["bar"] = None
    print(list(a.keys()))
```

### Expected Behavior

```
['foo', 'bar']
```

### Actual Behavior

```
['foo']
```

**Comment by plajjan** [comment]
@sydow you wanna have a look at this one? :)

**Comment by plajjan** [comment]
Discussed in today's meeting, #1659, and we've decided that we will switch over from using NULL for the internal use in our dicts to using some other valid pointer, so let's just define some new dummy pointer for this use. It should not be visible in the language, i.e. not go in `__builtin__.act` or so but rather be kept internally in like `dict.c`. I understood you will look into it, @sydow, right? :)

**Comment by sydow** [comment]
Fixed by #1661
